### ISSUE TITLE
Some values not displaying since Unity 6

### ISSUE THREAD

**Issue by PooperPig** [JohnBaracuda/Runtime-Monitoring#32]
I have recently moved to Unity 6 and noticed that in this project some value simply are not displayed.

Code below:
```

using Baracuda.Monitoring;
using UnityEngine;
using UnityEngine.Assertions;

/// <summary>
///     A replacement for KeepInLanes
///     Uses the simpler philosophy of moving either left or right while the _direction indicates so
///     Stops at the next lane when the _direction is 0
///     THE MIDDLE LANE IS LANE ZERO
///     Unless we have an even number of lanes which, I suppose, is possible!
/// </summary>
public class LaneMover : MonoBehaviour
{
#if DEBUG
    [SerializeField] protected bool _enableMonitoring;
#endif
    /// <summary>
    ///     The direction being moved in now.
    ///     This will be changed to _nextDirection when a lane is reached
    /// </summary>
#if DEBUG
    [Monitor] [MShowIf(nameof(_enableMonitoring))]
#endif
    private int _currentDirection;

#if DEBUG
    [Monitor] [MShowIf(nameof(_enableMonitoring))]
#endif
    private int _currentLane;
#if DEBUG
    [Monitor] [MShowIf(nameof(_enableMonitoring))]
#endif
    private float _laneWidth;
#if DEBUG
    [Monitor] [MShowIf(nameof(_enableMonitoring))]
#endif
    private int _maxLane;
#if DEBUG
    [Monitor] [MShowIf(nameof(_enableMonitoring))]
#endif
    private int _minLane;

#if DEBUG
    [Monitor] [MShowIf(nameof(_enableMonitoring))]
#endif
    private int _nextDirection;
#if DEBUG
    [Monitor] [MShowIf(nameof(_enableMonitoring))]
#endif
    private int _numberOfLanes;

    private Rigidbody _rigidBody;

    private void Start()
    {
#if DEBUG
        this.StartMonitoring();
#endif

        _rigidBody = GetComponent<Rigidbody>();
        Assert.IsNotNull(_rigidBody, $"Missing rigidbody on {gameObject.name}");
        _numberOfLanes = WorldState.Instance.NumberOfLanes;
        Assert.IsTrue(_numberOfLanes > 0, "Number of lanes must be greater than 0");

        // I'm assuming odd lanes for now
        _minLane = 0 - _numberOfLanes / 2;
        _maxLane = _numberOfLanes / 2;

        _laneWidth = (WorldState.Instance.TerrainWidth - WorldState.Instance.GutterWidth * 2) / _numberOfLanes;
    }

    private void FixedUpdate()
    {
        HandleLaneUpdates();
    }

    private void OnDestroy()
    {
#if DEBUG
        this.StopMonitoring();
#endif
    }
```

Screenshot of output shows that Min Lane displays as -1, lane Width as 4.00 and CurrentDirection as 0 but all other fields do not display a value.

I can't easily figure a commonality between those that do and don't display.

_minLane and _maxLane for example are adjacent in their declaration, and assignation but only one value is showing.

![image](https://github.com/user-attachments/assets/41995727-4563-4462-9216-7632a174bbc6)

**Comment by JohnBaracuda** [comment]
Hey, if there are no errors in the console I assume it's due to an IMGUI issue. You can try using the TextMeshPro UI adapter. I've been using the package in Unity 6 with the TextMeshPro adapter without any issues (so far). Let me know if this temporarily fixes the issue, as I'm not sure when I'll have the time to investigate and fix it.

Just a quick heads up, using [SerializeField] in a conditional like DEBUG or UNITY_EDITOR will cause deserialization issues in a build because those fields are excluded during runtime outside of the editor. Debugging this for the first time can be painful, just wanting to let you know because I've been there .

**Comment by PooperPig** [comment]
Thanks for the quick response.
Not sure what you mean by "using the TextMeshPro UI adapter"?
I just discovered the Monitoring settings for the first time - and found prefixing with the class name displayed all but one of the values in my example!

Thanks for the heads-up re [SerialiseField] in conditionals!

**Comment by JohnBaracuda** [comment]
So if you have Text Mesh Pro in your project you can import the optional Text Mesh Pro Resources from the Runtime Monitoring samples in the Package Manager. This will let you use Text Mesh Pro for the Monitoring UI instead of IMGUI (which is an outdated system) here is a link to the documation: [Text Mesh Pro](https://github.com/JohnBaracuda/com.baracuda.runtime-monitoring?tab=readme-ov-file#customized-setup)

**Comment by PooperPig** [comment]
Thanks for your help. UNfortunately the documentation just doesn't seem to make sense to me :(
When I go to the Runtime Monitoring settings it looks nothing like the one in the screenshot:
I see General, Filtering, Formatting, Debug, Assembly Settings, IL2CPP Settings and Documentation & LInks
No UIController setting.
Then the documentation says "Depending on the Unity version and your preferences, import and (sic) optional UIController package." 
But import from where?
Then "Set the prefab as the active UI controller" - but what prefab?

I am evidently missing something obvious!

Would love to try with textmeshPro just to see if it fixes my issue - but I'm stumped!

**Comment by JohnBaracuda** [comment]
The documentation appears to be outdated in that regard. You can import the samples here.

![image](https://github.com/user-attachments/assets/067663e2-5c21-4a58-8746-f1e240b7b141)

This will import the resources into the samples folder.

![image](https://github.com/user-attachments/assets/cae2f146-ce4d-4c25-9baa-4a84eb386d53)

And then in the settings just assign the prefab to the Monitoring UI Override field.

![image](https://github.com/user-attachments/assets/ebb9c253-06da-4265-a1ca-05898443fef0)

I hope this helps :)

**Comment by PooperPig** [comment]
I really appreciate your help with this.
I understood what needed to be done now - and it was fine until I download the Samples.
The MonitoringUI-TextMeshPro shows "Prefab has missing scripts:" when I select it
If I open the prefab, though, I can't see a missing script.
I can't drag the prefab to the Override because of this.
If I add an emptygame object to the scene and try to drag the UIControllerComponents script to it, I get the error
"Can't add script component UIControllerComponents because the script class cannot be found"
It seems to be the same for any script I create in that folder!
After a deal of R&D - and more than a little clutching at straws - I moved all of the sources up a couple of levels in the folder hierarchy - and lo! it works.
And yes - it fixed the problem with missing values!
Out of interest my whole path was
C:\Users\xxxxx\Projects\xxxxxxxxxxxxxxxxxxxxx\xxxxxxxxxGame\Assets\Samples\Runtime Monitoring\3.6.0\Text Mesh Pro Resources\Prefabs

without the x obviously!

Which is about 132 characters - which takes the total length to 140 which is (I have discovered now!) the maximum for Unity!

tl;dr; Your instructions worked, it fixed my issue, I am happy!
And I've learned something new about maximum file lengths in Unity :)

A coffee or two are on there way via PayPal.

**Comment by JohnBaracuda** [comment]
I'm glad you got it working! File path limits are pain, been there myself. Happy to have helped, and thanks for the coffee ;)
